lib/fauna/query.rb

```diff
@@ -69,11 +69,38 @@ def self.quote(expr)
       { quote: expr }
     end
 
+    Thread.current[:fauna_lambda_var_name] = 'a'
+
+    ##
+    # A lambda expression
+    # Reference: {FaunaDB Basic Forms}[https://faunadb.com/documentation/queries#basic_forms]
+    #
+    # This form generates the names of lambda parameters for you, and is called like:
+    #
+    #   query.lambda do |a|
+    #     query.add(a, a)
+    #   end
+    #   # Produces: {lambda: 'a', expr: {add: [{var: 'a'}, {var: 'a'}]}}
+    #
+    # You can also use ::lambda_expr and ::var directly.
+    def self.lambda
+      var_name = Thread.current[:fauna_lambda_var_name]
+      # Move var name for inner closures from 'a' to 'b', or 'b' to 'c', etc.
+      Thread.current[:fauna_lambda_var_name] =
+        [Thread.current[:fauna_lambda_var_name].codepoints.first + 1].pack 'U'
```

Comment:
Can you change this to something like picking a character from an array, instead of incrementing the unicode character?